### ISSUE TITLE
Use Java config code snippets in transaction management section

### ISSUE THREAD

**Issue by abccbaandy** [spring-projects/spring-framework#34093]
Many example in the reference still use xml format, but most project doesn't use xml any more.

Can it update to java config?

For ex:
Transaction Management
https://docs.spring.io/spring-framework/reference/data-access/transaction.html

**Comment by pratyakshTrivedi219** [comment]
Interested to make the necessary changes to the transaction management section, by updating the XML format examples and replacing them with java config examples. Here is the updated file

[strategies.txt](https://github.com/user-attachments/files/19367839/strategies.txt)

**Comment by jisub-dev** [comment]
I can help with this. I'll create Java Config examples for transaction management to replace XML examples.

**Comment by sbrannen** [comment]
Hi @jisub-dev,

Feel free to submit a PR for review. 👍

**Comment by sbrannen** [comment]
> Hi [@jisub-dev](https://github.com/jisub-dev),
> 
> Feel free to submit a PR for review. 👍

Though, we need to decide what to do about the existing XML examples.

It may suffice to have Java config and XML in _tabs_ side-by-side.

@sdeleuze, perhaps you can provide insight on any plans you may have had for this task.

**Comment by sdeleuze** [comment]
Indeed, providing programmatic and XML configurations side by side is the way to go.
I will provide a review for the related PR that only covers a subset of what should be done.

**Comment by sdeleuze** [comment]
Superseded by #36323.
